Bug-Fix for Version 3.0.1 (#632)
VDA5050/VDA5050: VDA5050_EN.md

Reviewer: typo
```suggestion
The `requestStatus` is set to REQUESTED and the combination of `edgeId` and `sequenceId` references the edge's trajectory the robot asks to deviate from. The mobile robot has the option to request the approval for several edges simultaneously as long as they are part of its current base. The usage of each corridor shall be requested in a dedicated `edgeRequest` and each request shall be approved individually by fleet control via the `response` topic (see Section [6.9 Request/response mechanism](#69-requestresponse-mechanism)).
```

Reviewer 2: Updated to fix faulty words

Reviewer: typo
```suggestion
An extensible enum includes but is not limited to the predefined values for the parameter.
```

Reviewer: double the
```suggestion
*orientation* | rad | float64 | Orientation of the mobile robot on the trajectory of the edge. The value `orientationType` defines whether it shall be interpreted relative to the global project-specific map coordinate system or tangential to the trajectory of the edge. In case of tangential to the trajectory, 0.0 denotes driving forwards and PI denotes driving backwards. <br>Example: orientation Pi/2 rad may lead to a rotation of 90 degrees.<br><br>If the mobile robot starts in a different orientation, and if `reachOrientationBeforeEntering` is set to "false", rotate the mobile robot on the edge to the desired orientation.<br>If `reachOrientationBeforeEntering` is "true", rotate before entering the edge.<br>If this is not possible, the order shall be rejected.<br><br>If no trajectory is defined, apply the orientation and any rotation to the direct path between the two connecting nodes of the edge.<br>If no orientation is defined, the mobile robot may assume any orientation on the edge.
```

Reviewer: 2.x leftover
```suggestion
**actionStates [actionState]** | | array | Contains an array of all actions from the current order. The action states are kept as long as the order remains active and cleared when accepting a new order. <br>This may include actions from previous nodes, that are still in progress.<br><br>When an action is completed, an updated state message is published with `actionStatus` set to 'FINISHED' and if applicable with the corresponding `actionResult`.
```

Reviewer: align heading level

```suggestion
### 6.4.3 Communication for interactive zones 
```

Reviewer: fix anchor

```suggestion
Actions that are to be executed on an edge shall only run while the mobile robot is on the edge (see Section [6.6.2 Traversal of nodes and edges](#662-traversal-of-nodes-and-edges)).
```

Reviewer: Fix anchor

```suggestion
***allowedDeviationXY*** | m | JSON object | Indicates how precisely a mobile robot shall match the position of a node for it to be considered traversed.<br>(see also Section [6.1.3 Order cancellation](#613-order-cancellation) and [6.6.2 Traversal of nodes and edges](#662-traversal-of-nodes-and-edges)).
```

Reviewer: SerialNumber -> serialNumber
```suggestion
| &emsp;*maximumTopicSerialLength* | uint32 | Maximum length of serial number part in MQTT-topics.<br><br>Affected parameters:<br>order.serialNumber<br>instantActions.serialNumber<br>state.serialNumber<br>visualization.serialNumber<br>connection.serialNumber<br>zoneSet.serialNumber<br>response.serialNumber |
```

Reviewer: reponses -> responses
```suggestion
| *leaseExpiry* <br><br> } | string | Timestamp (ISO 8601, UTC); YYYY-MM-DDTHH:mm:ss.fffZ (e.g.“2017-04-15T11:40:03.123Z”). A timestamp for the release to expire shall only be sent with responses granting a request.
```

Reviewer: As we have a default defined here ("If not defined, ... STOP") - should this be an optional parameter?

Reviewer 2: It definitely was designed as an optional parameter. We should either make it optional or remove the default setting.

Reviewer: Missing actions
```suggestion
'INSTANT_ACTION_STATES_FULL' | 'URGENT' | Array of `instantActionStates` is becoming too long to manage | - | Until instantActionStates are cleared by fleet control.
ZONE_ACTION_STATES_FULL' | 'URGENT' | Array of `zoneActionStates` is becoming too long to manage | - | Until instantActionStates are cleared by fleet control.

```

Reviewer 2: *Missing Error types

Reviewer: INVALID_ORDER -> INVALID_ORDER_ACTION   (former does not exist anymore)
```suggestion
errorType | | string | Error type, extensible enumeration including the following predefined values <br>Enum {'UNSUPPORTED_PARAMETER', 'NO_ORDER_TO_CANCEL', 'VALIDATION_FAILURE', 'INVALID_ORDER_ACTION', 'OUTDATED_ORDER_UPDATE', 'SAME_ORDER_UPDATE_ID', 'ORDER_UPDATE_FOLLOWING_CANCEL', 'OUTSIDE_OF_CORRIDOR', 'DUPLICATE_MAP', 'DUPLICATE_ZONE_SET', 'BLOCKED_ZONE_VIOLATION', 'RELEASE_LOST', 'ZONE_ACTION_CONFLICT', 'NODE_UNREACHABLE', 'LOCALIZATION_ERROR', 'UNKNOWN_MAP_ID', ...}.
```

Reviewer: Missing column
```suggestion
startPause | - | Activation of the mode is in preparation.<br>If the mobile robot supports an instant transition, this state can be omitted. | - | Mobile robot is not moving. <br>All pauseable actions are paused. <br> The pause mode has been activated. <br>The mobile robot reports paused: "true". | The pause mode cannot be activated for some reason (e.g., overridden by hardware switch). | -
```

Reviewer: Add break to remove text from figure 10's caption  (matching 2.1 caption again)
```suggestion

The area in which the mobile robot is allowed to navigate independently (and deviate from the original edge trajectory) is defined by a left and a right boundary.
```

Reviewer: remove extra | 
```suggestion
**BLOCKED**|BLOCKED|BLOCKED|BLOCKED|BLOCKED|BLOCKED|BLOCKED|BLOCKED|BLOCKED|BLOCKED|BLOCKED|BLOCKED|BLOCKED
```

Reviewer: Add missing |
```suggestion
| | direction | float64 | Preferred direction of travel within the zone in radians. The direction of travel is the angular orientation of the mobile robot's velocity vector in the project-specific coordinate system.|
```

Reviewer: remove double parenthesis
```suggestion
***actionParameters [actionParameter]*** | | array | Array of actionParameter objects for the indicated action, e.g., "deviceId", "loadId", "external triggers". <br><br> An example implementation can be found in [7.3.1 Format of action parameters](#731-format-of-action-parameters).
```

Reviewer: missing blank
```suggestion
**zoneSet** { | | JSON object|
```

Reviewer: Wrong column
```suggestion
**nodeState** { | | JSON object |
```

Reviewer: Missing blank
```suggestion
| vertex {| JSON object| |
```

Reviewer: Missing line break
```suggestion
| ***versions[versionInfo]*** | array | Array of key-value pair objects containing software and hardware information.|
| { | | |
```

Reviewer: Typo
```suggestion
For permission to enter a 'COORDINATED_REPLANNING' zone with a planned path or for replanning its path within the zone, the `requestType` shall be set to 'REPLANNING'.
```

Reviewer: Typo
```suggestion
If a mobile robot navigates through a workspace on the map that is covered by two or more 'COORDINATED_REPLANNING' zones, it shall request its path within this area individually for each zone and receive approval from the fleet control before entering or changing paths.
```

Reviewer: missing 'zone'
```suggestion
| ***trajectory*** <br> } | object | Optional for 'COORDINATED_REPLANNING' zone requests only with the planned trajectory through the zone. |
```

Reviewer: typo
```suggestion
clearZoneActions | - | | - | The zone actions array has been cleaned from all FINISHED or FAILED zoneActions. | - | - 
```

Reviewer: typo
```suggestion
topic | string | Topic (e.g., order or state) see Section [4.3 Topics for Communication](#43-topics-for-communication)
```

Reviewer: lastNodeId is toplevel
```suggestion
initializePosition | - | Initializing of the new pose in progress (confidence checks, etc.). <br>If the mobile robot supports an instant transition, this state can be omitted. | - | The pose has been reset. <br>The mobile robot reports <br>mobileRobotPosition.x = x, <br>mobileRobotPosition.y = y, <br>mobileRobotPosition.theta = theta <br>mobileRobotPosition.mapId = mapId <br>lastNodeId = lastNodeId | The pose is not valid or cannot be reset. <br>General localization problems should correspond with an error. | -
```

Reviewer: singular
```suggestion
Some mobile robots cannot process zones at all, while other mobile robots might only be able to work with a certain subset of zone types, such as 'BLOCKED'. All mobile robots shall therefore report to fleet control which zones they are able to understand by adding the according zone names to the `supportedZones` array under `typeSpecification` in their factsheet.
```

Reviewer: loads plural;  move arrays to bottom
```suggestion
- Change in the `operatingMode` field
- Change in the `driving` field
- Change in the `paused` field
- Change in the `safetyState` object
- Change in the `newBaseRequest` field
- Change in the `lastNodeId` or `lastNodeSequenceId` field
- Change in the `edgeRequests` or `zoneRequests` arrays
- Change in the `powerSupply.charging` field
- Change in the `nodeStates` or `edgeStates` arrays
- Change in the `actionStates`, `instantActionStates` or `zoneActionStates` arrays
- Changes in the `loads` array
- Change in the `errors` array
- Change in the `zoneSets` array
- Change in the `maps` array
```

Reviewer: typo
```suggestion
When a mobile robot receives an `instantAction`, an appropriate `actionState` shall be added to the `instantActionStates` array of the mobile robot's state.
```

Reviewer: singular
```suggestion
| &emsp;*action.actionsParameters* | uint32 | Maximum number of parameters per action processable by the mobile robot. |
```

Reviewer: remove object name from data type column
```suggestion
***zoneSets[zoneSet]*** | | array | Array of zoneSet objects that are currently stored on the mobile robot.
```

Reviewer: missing unit
```suggestion
theta | rad | float64 | Range: [-Pi ... Pi]<br><br>Orientation of the mobile robot.
```

Reviewer: double deleted
```suggestion
deleteZoneSet | - | Mobile robot deletes the zone set with requested zoneSetId from its internal memory. | - | The zone set has been deleted. The mobile robot removes zoneSet object from its state. | The zone set could not be deleted, e.g., because zone set is currently in use or the requested zone set has already been deleted before. | -
```

Reviewer: typo
```suggestion
As long as the actions of an order are not in state 'FINISHED' or 'FAILED' the mobile robot shall not report operating mode 'MANUAL', 'SERVICE' or 'TEACH_IN'. `nodeStates` and `edgeStates` shall not be emptied before the operating mode 'MANUAL', 'SERVICE' or 'TEACH_IN' is reported.
```

Reviewer 2: Done

Reviewer: typo
```suggestion
In the following matrix possible interactions between zones are described. The matrix is symmetric, as the interaction between two zones is the same, regardless of the order in which they are considered. For each combination, there is either a zone behavior that is overruling the other (e.g., a 'BLOCKED' zone overrules a 'LINE_GUIDED' zone) or there is no conflict (e.g., a 'LINE_GUIDED' zone and a 'COORDINATED_REPLANNING' zone). 'DIRECTED' and 'BIDIRECTED' zones shall not overlap, since this might lead to an undefined behavior. The column No Zone defines the behavior for contour-based zones, where mobile robots can be inside a defined zone type and an area without a zone at the same time. For kinematic center-based zones the mobile robot can only be completely within or outside the zone, so there is no possible interaction.
```

Reviewer: typo
```suggestion
| | bidirectedLimitation | string | Enum {'SOFT', 'RESTRICTED'}<br>SOFT: Mobile robots may deviate from the defined directions of travel, but should avoid it, RESTRICTED: The mobile robot shall not traverse in any other direction than the directions of travel, except for obstacle avoidance. |
```

Reviewer: remove germanification
```suggestion
|&emsp; *loadPositions* | array of string | Array of load positions or load handling devices, this load set is valid for.<br>*If this parameter does not exist or is empty, this load set is valid for all load handling devices on this mobile robot.* |
```

Reviewer: We should have a second look here.
Caption only names `requestStatus`, but the ENUM values are mixed from both requestStatus and grantType - e.g., QUEUED is grantType only.
REJECTED is also missing in this list

Reviewer: Should LINE_GUIDEDxBIDIRECTED resolve to LINE_GUIDED (same as it is for "DIRECTED")?

Reviewer 2: Moved to Issue #641

Reviewer: Actions 'trigger', 'retry', 'skipRetry' are part of table 4, but not table 5. 
Should we add them?

Reviewer 2: See PR #640

Reviewer: Remove ambiguous description part.
```suggestion
zoneSetId | | string | Unique identifier of the zone set.
```

Reviewer: “ -> "
```suggestion
charging | | boolean | "true": charging in progress.<br>"false": the mobile robot is currently not charging. Shall only be reported as "false" if the robot is available to take orders.
```

Reviewer: cirrect to align with description above
```suggestion
MANUAL | NO | YES | YES | YES, if start of new order not possible anymore | YES | NO | NO
```

Reviewer: Should we also define usage of the predefined error here?

Reviewer: leftover
```suggestion
INTERVENED | Fleet control is not in control of the mobile robot. The mobile robot is reporting its state correctly.<br>HMI can be used to control the steering, velocity and handling devices of the mobile robot.<br>Fleet control is allowed to send orders or order updates to the mobile robot to be executed after changing back into operating mode 'AUTOMATIC' or 'SEMIAUTOMATIC'. Fleet control shall not send any instant action except `cancelOrder`.<br>The mobile robot shall not clear the order but shall remove all zone requests from the state, also if the mobile robot is already inside a 'RELEASE' zone. (*Remark: If necessary, the fleet control can continue to track the position of the mobile robot and decide whether clearance for other mobile robots is possible.*) The mobile robot shall not request any permissions to enter a 'RELEASE' zone or for replanning inside a 'COORDINATED_REPLANNING' zone.<br>If entering operating mode 'INTERVENED' has any impact on running actions the mobile robot shall reflect this in the state message accordingly.<br>If the mobile robot leaves this operating mode and does not directly switch into 'AUTOMATIC' or 'SEMIAUTOMATIC' mode it shall act according to new operating mode. If the mobile robot leaves this operating mode and switches directly into 'AUTOMATIC' or 'SEMIAUTOMATIC' mode the mobile robot shall continue executing any current order. If the mobile robot detects during operating mode 'INTERVENED' that a continuation of the current order is not possible the mobile robot shall switch into operating mode 'MANUAL' and act accordingly.
```

Reviewer 2: SEMI-AUTOMATIC -> SEMIAUTOMATIC

Reviewer: plural
```suggestion
## 7.5 Implementation of the responses message
```

Reviewer: plural
```suggestion
  [7.5 Implementation of the responses message](#75-implementation-of-the-responses-message)<br>
```

Reviewer: backticks
```suggestion
1. Mobile robot sends "vda5050/v3/manufacturer/serialNumber/connection" with `connectionState` set to 'OFFLINE'.
```

Reviewer: typo
```suggestion
'PAUSED' | The action is paused because of a startPause instantAction or external trigger (pause button on the mobile robot)
```

Reviewer: there is no information message
```suggestion
It is up to the mobile robot to decide how long it reports this information.
```

Reviewer: backticks
```suggestion
The order of the nodes and edges within those lists also governs the sequence in which the nodes and edges shall be traversed. The `sequenceId` is shared between nodes and edges and defines the sequence of traversal. The first node has a `sequenceId` of 0, the first edge has a `sequenceId` of 1, the second node has a `sequenceId` of 2, etc. An edge with `sequenceId` n connects the nodes with `sequenceId` n-1 and n+1. The `sequenceId` shall be continuous within an order.
```

Reviewer: Align with other listings

```suggestion
|*blockingTypes* | array of enum | Array of possible blocking types for defined action. <br> Enum {'NONE', 'SINGLE', 'SOFT', 'HARD'} |
```

Reviewer: quotes
```suggestion
| mobileRobotClass | string | Simplified description of the mobile robot class.<br> Extensible enum: {'FORKLIFT', 'CONVEYOR', 'TUGGER', 'CARRIER', ...}<br>'FORKLIFT': forklift,<br>'CONVEYOR': Mobile robot with conveyors on it,<br>'TUGGER': tugger,<br>CARRIER: load carrier with or without lifting unit. |
```

Reviewer: quotes
```suggestion
zoneSetStatus <br>}| | string | Enum {'ENABLED', 'DISABLED'}<br>'ENABLED': Indicates this zone set is currently actively used on the mobile robot. At most one zone set for each map can have its status set to 'ENABLED' .<br>'DISABLED': Indicates this zone set is currently not enabled on the mobile robot and thus could be enabled or deleted by fleet control.
```

Reviewer: quotes
```suggestion
| requestType | enum | Enum {'CORRIDOR'}<br> Enum specifying the type of request. Set to 'CORRIDOR' if requesting to deviate from the predefined trajectory within the defined work space. |
```

Reviewer 2: In theory, there are no quotation marks in tables. However, this is currently inconsistent,

Reviewer: data type
```suggestion
| requestStatus | enum | Enum {'REQUESTED', 'GRANTED', 'REVOKED', 'EXPIRED'}<br>When stating a request, this is set to 'REQUESTED'. After response or update from fleet control set to 'GRANTED' or 'REVOKED'. If lease time expires set to 'EXPIRED'.|
```

Reviewer 2: Moved to PR #642, currently enums are either `string` or `enum`, should now be string everywhere.

Reviewer: data type

```suggestion
| requestType | enum | Enum {'ACCESS', 'REPLANNING'}<br>Specifying the type of zone the request relates to. Feasible values are 'ACCESS' or 'REPLANNING'. |
```

Reviewer 2: enum is still a sting. has to be fixed at other places too.

Reviewer: bold italic

```suggestion
***infoReferences [infoReference]*** | | array | Array of references.
```

Reviewer: singular entry
```suggestion
| **optionalParameters** [**optionalParameter**] | array | Array of supported and/or required optional parameters.<br>Optional parameters that are not listed here are assumed to be not supported by the mobile robot. |
```

Reviewer: h4
```suggestion
#### mobileRobotGeometry
```

Reviewer: indentation missing
```suggestion
|&emsp;*actionResult* | string | Free-form text: description of the result. |
|&emsp;*blockingTypes* | array of enum | Array of possible blocking types for defined action. <br> Enum {'NONE', 'SOFT', 'SINGLE', 'HARD'} |
|&emsp;pauseAllowed | boolean | "true": action can be paused via startPause, "false": action cannot be paused. |
|&emsp;cancelAllowed | boolean | "true": action can be cancelled via cancelOrder, "false": action cannot be cancelled. |
```

Reviewer: indentation missing
```suggestion
| &emsp;*criticalLowChargingLevel* | float64 | Specifies the critical charging level in percent at or below which the fleet control should only send orders that command the mobile robot to a charging station. |
| &emsp;*maximumDesiredChargingLevel* | float64 | Specifies the maximum desired charging level in percent. |
| &emsp;*minimumDesiredChargingLevel* | float64 | Specifies the minimum desired charging level in percent. |
| &emsp;*minimumChargingTime* | uint32 | Specifies the desired minimum charging time in seconds. |
```

Reviewer: Match text for same field in visualization
```suggestion
***velocity*** | | JSON object | The mobile robot velocity in mobile robot coordinates.
```

Reviewer: reference

```suggestion
actionType | | string | Name of action as described in the first column of table 4. <br> Identifies the function of the action.
```

Reviewer: plural
```suggestion
Potential conflicts in orders due to overlapping of zones or combination of zone and edge properties and how to resolve them are addressed in section [6.4.4 Interactions between zones](#644-interactions-between-zones). For released nodes that are part of the order but are restricted due to zones (e.g., node located within a 'BLOCKED' or 'RELEASE' zone), the robot is expected to act according to the zones (e.g., not enter or wait for 'GRANTED' state of the request).
```

Reviewer: Plural
```suggestion
    [6.4.4 Interactions between zones](#644-interactions-between-zones)<br>
```

Reviewer: Switch
```suggestion
    [6.1.2 Order and order updates](#612-order-and-order-updates)<br>
```

Reviewer: Switch
```suggestion
### 6.1.2 Order and order updates
```

Reviewer: Match caption
```suggestion
    [6.3.2 Maps in the mobile robot state](#632-maps-in-the-mobile-robot-state)<br>
```

Reviewer: typo
```suggestion
>Figure 20 - allowedDeviationXY ellipse
```

Reviewer: typo? not sure here
```suggestion
Reprint and any other form of reproduction is only permitted with specification of the source.
```

Reviewer: grammar
```suggestion
A driverless system for material transport primarily in operational settings, controlled by automation independently of its level of autonomy [Source ISO 3691-4]
```

Reviewer: grammar
```suggestion
For a local broker the MQTT topic levels are suggested as follows:
```

Reviewer: grammar
```suggestion
There are at least the following participants for the operation of a driverless transport system:
```

Reviewer: missing the
```suggestion
Is `orderId` of the received order different to `orderId` of the order the mobile robot currently holds?
```

Reviewer: grammar
```suggestion
10) **Is the received update a valid continuation of the previously completed order?**: Is the first node of the received order the current decision point according to the order update chapter? The mobile robot is not executing any actions anymore nor is it waiting for a continuation of the order (meaning that it has completed its base with all related actions and does not have a horizon). In this case, the order update is only accepted if the first node of the new base is equal to the last node of the previous base.
```

Reviewer: dash
```suggestion
>Figure 5 - Pseudocode of an order update. Note the change of the `orderUpdateId`.
```

Reviewer: dash
```suggestion
>Figure 4 - Pseudocode of an order.
```

Reviewer: order mode -> operating mode
```suggestion
3. The warning shall be reported until the mobile robot is in an operating mode that allows for new orders.
```

Reviewer: quote + period
```suggestion
2. The mobile robot shall report an error of type 'VALIDATION_FAILURE' and level 'WARNING'.
```

Reviewer: remove trailing br
```suggestion
enableZoneSet | - | Enable a previously downloaded zone set explicitly to be used in orders. | yes | zoneSetId (string) | zoneSets | yes | yes | no | no
```

Reviewer: remove leading br
```suggestion
cancelOrder | - | Mobile robot is stopping or driving, until it reaches the next node. | - | Mobile robot is not moving. Mobile robot has canceled executing the order and is in idle state. | Mobile robot has no active order.<br>The previous order has already been canceled.<br>Passed orderId does not match the currently active orderId. | -
```

Reviewer: grammar
```suggestion
To ensure consistent navigation among different types of mobile robots, the position is always specified in reference to the project-specific coordinate system (see Figure 12). The project-specific coordinate system refers to the coordinate system that is defined for the interaction between fleet control and the mobile robot.
```

Reviewer: cap
```suggestion
1. **Fleet control enables map**: Use the `enableMap` instant action to set a map to 'ENABLED' on the mobile robot. Other versions of the same `mapId` with different `mapVersion` are set to 'DISABLED'.
```

Reviewer: missing STARTUP
```suggestion
As long as the actions of an order are not in state 'FINISHED' or 'FAILED' the mobile robot shall not report operating mode 'MANUAL', 'SERVICE' or 'TEACH_IN'. `nodeStates` and `edgeStates` shall not be emptied before the operating mode 'MANUAL', 'SERVICE', 'STARTUP' or 'TEACH_IN' is reported.
```

Reviewer: plural
```suggestion
- The values of `orderId`, `orderUpdateId`, `lastNodeId` and `lastNodeSequenceId` remain unchanged.
```

Reviewer: remove "to"
```suggestion
If the mobile robot cannot process the order due to an unsupported parameter, it shall communicate this with an error of type 'UNSUPPORTED_PARAMETER' and error level 'CRITICAL' and reject the order.
```

Reviewer: grammar
```suggestion
actionId | | string | Unique ID to identify the action and map it to the `actionState` in the state. <br>Suggestion: Use UUID.
```

Reviewer: double space
```suggestion
| *maximumSpeed* | float64 | Required only for SPEED_LIMIT zone as defined in chapter [6.4.1 Zone types](#641-zone-types).| 
```

Reviewer: remove "the"
```suggestion
localized | | boolean | "true": Mobile robot is localized. `x`, `y`, and `theta` can be trusted.<br>"false": Mobile robot is not localized. `x`, `y`, and `theta` cannot be trusted.<br>Changing the state to "false" shall only happen if the mobile robot cannot determine its position anymore. The mobile robot shall report this state via an error (`errorType` = 'LOCALIZATION_ERROR', `errorLevel` = 'FATAL'). While this is set to "false", the mobile robot shall not resume automatic driving or continue its order.
```

Reviewer: add unit
```suggestion
x | m | float64 | X-coordinate of the point of reference.
y | m | float64 | Y-coordinate of the point of reference.
z | m | float64 | Z-coordinate of the point of reference.
```

Reviewer: grammar
```suggestion
*theta*<br> } | | float64 | Orientation of the load's bounding box. <br>Important for tuggers, trains, etc.
```

Reviewer: missing blanks
```suggestion
fieldViolation<br>} | | boolean | Protective field violation (e.g., by laser or bumper).<br>"true": field is violated<br>"false": field is not violated.
```

Reviewer: close ul
```suggestion
| &emsp;*description*| string | Free-form text: description of optional parameter, e.g., <ul><li>Reason, why the optional parameter direction is necessary for this mobile robot type and which values it can contain.</li><li>The parameter nodeMarker shall contain unsigned integers only.</li><li>NURBS support is limited to straight lines and circle segments.</li></ul>|
```

Reviewer: missing blank
```suggestion
| zoneSet { | JSON object | Zone set detailing a dedicated map. | 
```

---|---|---|---|---|---|---|---|---|---
startPause | stopPause | Activates the pause mode. <br>A linked state is required, because many mobile robots can be paused by using a hardware switch. <br>No more automatic driving - reaching next node is not necessary. Actions that can be paused (`pauseAllowed`=`true`), shall be paused, other actions continue. Order execution is resumed after stopPause. | yes | - | paused | yes | no | no | no
stopPause | startPause | Deactivates the pause mode. <br>Movement and all other actions will be resumed (if any). <br>A linked state is required because many mobile robots can be paused by using a hardware switch. <br>stopPause can also restart mobile robots that were stopped with a hardware button that triggered startPause (if configured). | yes | - | paused | yes | no | no | no

> startHibernation | stopHibernation | Initiates hibernate mode, in which the mobile robot shall remain connected to the MQTT broker but no longer needs to send state messages. The mobile robot shall report this action as 'FINISHED' before discontinuing publishing state messages and publish a connection state of 'HIBERNATING'. If the mobile robot has an active order, it shall clear it. Reaching the next node is not required. <br>While in 'HIBERNATING' connection state, mobile robot shall not be moving. The mobile robot shall only receive and respond to the instant action 'stopHibernation' and shall not respond to any other commands, such as orders or additional instant actions. <br>If the mobile robot's battery becomes critically low while in this mode, the mobile robot may stop 'HIBERNATING' autonomously to report an error. In case a wake‑up time is set, the mobile robot is able to autonomously exit the 'HIBERNATING' connection state at the specified time and will publish the corresponding connection state transition before resuming normal operation. Parameter wakeUpTime is a timestamp (ISO 8601, UTC); YYYY-MM-DDTHH:mm:ss.fffZ (e.g., "2017-04-15T11:40:03.123Z"). | yes | wakeUpTime (string, optional) | - | yes | no | no

Reviewer: missing column
```suggestion
startHibernation | stopHibernation | Initiates hibernate mode, in which the mobile robot shall remain connected to the MQTT broker but no longer needs to send state messages. The mobile robot shall report this action as 'FINISHED' before discontinuing publishing state messages and publish a connection state of 'HIBERNATING'. If the mobile robot has an active order, it shall clear it. Reaching the next node is not required. <br>While in 'HIBERNATING' connection state, mobile robot shall not be moving. The mobile robot shall only receive and respond to the instant action 'stopHibernation' and shall not respond to any other commands, such as orders or additional instant actions. <br>If the mobile robot's battery becomes critically low while in this mode, the mobile robot may stop 'HIBERNATING' autonomously to report an error. In case a wake‑up time is set, the mobile robot is able to autonomously exit the 'HIBERNATING' connection state at the specified time and will publish the corresponding connection state transition before resuming normal operation. Parameter wakeUpTime is a timestamp (ISO 8601, UTC); YYYY-MM-DDTHH:mm:ss.fffZ (e.g., "2017-04-15T11:40:03.123Z"). | yes | wakeUpTime (string, optional) | - | yes | no | no | no
```

> stopHibernation | startHibernation | Ends hibernate mode. To initiate wake‑up while the mobile robot is in the 'HIBERNATING' state, a control device (onboard or external) shall subscribe to the `instantAction` topic and remain connected to the MQTT broker. Because the mobile robots standard control device may be partially shut down during hibernation, the wake‑up may be triggered by a distinct MQTT client (separate from the mobile robots usual communication client).<br>Upon success, the mobile robot shall publish the connection state ONLINE.| yes | - | - | yes | no | no

Reviewer: missing column; topic name
```suggestion
stopHibernation | startHibernation | Ends hibernate mode. To initiate wake‑up while the mobile robot is in the 'HIBERNATING' state, a control device (onboard or external) shall subscribe to the `instantActions` topic and remain connected to the MQTT broker. Because the mobile robots standard control device may be partially shut down during hibernation, the wake‑up may be triggered by a distinct MQTT client (separate from the mobile robots usual communication client).<br>Upon success, the mobile robot shall publish the connection state ONLINE.| yes | - | - | yes | no | no | no
```

---|---|---|---|---|---|---|---|---|---
startPause | stopPause | Activates the pause mode. <br>A linked state is required, because many mobile robots can be paused by using a hardware switch. <br>No more automatic driving - reaching next node is not necessary. Actions that can be paused (`pauseAllowed`=`true`), shall be paused, other actions continue. Order execution is resumed after stopPause. | yes | - | paused | yes | no | no | no
stopPause | startPause | Deactivates the pause mode. <br>Movement and all other actions will be resumed (if any). <br>A linked state is required because many mobile robots can be paused by using a hardware switch. <br>stopPause can also restart mobile robots that were stopped with a hardware button that triggered startPause (if configured). | yes | - | paused | yes | no | no | no

> startHibernation | stopHibernation | Initiates hibernate mode, in which the mobile robot shall remain connected to the MQTT broker but no longer needs to send state messages. The mobile robot shall report this action as 'FINISHED' before discontinuing publishing state messages and publish a connection state of 'HIBERNATING'. If the mobile robot has an active order, it shall clear it. Reaching the next node is not required. <br>While in 'HIBERNATING' connection state, mobile robot shall not be moving. The mobile robot shall only receive and respond to the instant action 'stopHibernation' and shall not respond to any other commands, such as orders or additional instant actions. <br>If the mobile robot's battery becomes critically low while in this mode, the mobile robot may stop 'HIBERNATING' autonomously to report an error. In case a wake‑up time is set, the mobile robot is able to autonomously exit the 'HIBERNATING' connection state at the specified time and will publish the corresponding connection state transition before resuming normal operation. Parameter wakeUpTime is a timestamp (ISO 8601, UTC); YYYY-MM-DDTHH:mm:ss.fffZ (e.g., "2017-04-15T11:40:03.123Z"). | yes | wakeUpTime (string, optional) | - | yes | no | no | no

Reviewer 3: Back quotes for the camelcased wakeUpTime parameter or no camelcasing "wake-up time"?

```suggestion
startHibernation | stopHibernation | Initiates hibernate mode, in which the mobile robot shall remain connected to the MQTT broker but no longer needs to send state messages. The mobile robot shall report this action as 'FINISHED' before discontinuing publishing state messages and publish a connection state of 'HIBERNATING'. If the mobile robot has an active order, it shall clear it. Reaching the next node is not required. <br>While in 'HIBERNATING' connection state, mobile robot shall not be moving. The mobile robot shall only receive and respond to the instant action 'stopHibernation' and shall not respond to any other commands, such as orders or additional instant actions. <br>If the mobile robot's battery becomes critically low while in this mode, the mobile robot may stop 'HIBERNATING' autonomously to report an error. In case a wake‑up time is set, the mobile robot is able to autonomously exit the 'HIBERNATING' connection state at the specified time and will publish the corresponding connection state transition before resuming normal operation. Parameter `wakeUpTime` is a timestamp (ISO 8601, UTC); YYYY-MM-DDTHH:mm:ss.fffZ (e.g., "2017-04-15T11:40:03.123Z"). | yes | wakeUpTime (string, optional) | - | yes | no | no | no
```

enableZoneSet | - | Enable a previously downloaded zone set explicitly to be used in orders. | yes | zoneSetId (string) | zoneSets | yes | yes | no | no
deleteZoneSet | - | Trigger the removal of a zone set from the mobile robot's memory. | yes | zoneSetId (string) | zoneSets | yes | no | no | no
clearInstantActions | - | Removes all finished or failed instant actions from the mobile robot state. | yes | - | instantActionStates | yes | yes | no | no
clearZoneActions | - | Removes all finished or failed zone actions from the mobile robot's state. | yes | - | zoneActionStates | yes | yes | no | no
stateRequest | - | Requests the mobile robot to send a new state message. | yes | - | - | yes | no | no | no
logReport | - | Requests the mobile robot to generate and store a log report. | yes | reason<br>(string) | - | yes | no | no | no

> pick | drop<br><br>(if automated) | Request the mobile robot to pick a load. <br>Mobile robots with multiple load handling devices can process multiple pick operations in parallel. <br>In this case, the parameter loadHandlingDevice needs to be present (e.g., LHD1). <br>The parameter stationType informs how the pick operation is handled in detail (e.g., floor location, rack location, passive conveyor, active conveyor, etc.). <br>The load type informs about the load unit and can be used to switch field for example (e.g., EPAL, INDU, etc). <br>For preparing the load handling device (e.g., pre-lift operations based on the height parameter), the action could be announced in the horizon in advance. <br>But, pre-Lift operations, etc., are not reported as 'RUNNING' in the mobile robot state, because the associated node is not released yet.<br>If on an edge, the mobile robot can use its sensing device to detect the position for picking the node. | no | loadHandlingDevice (string, optional)<br>stationType (string, optional)<br>stationName (string, optional)<br>loadType (string, optional) <br>loadId (string, optional)<br>height (float64, optional)<br>defines bottom of the load related to the floor<br>depth (float64, optional) for forklifts<br>side (string, optional) e.g., conveyor | loads | no | yes | yes | no

Reviewer 3: ```suggestion
pick | drop<br><br>(if automated) | Request the mobile robot to pick a load. <br>Mobile robots with multiple load handling devices can process multiple pick operations in parallel. <br>In this case, the parameter `loadHandlingDevice` needs to be present (e.g., LHD1). <br>The parameter `stationType` informs how the pick operation is handled in detail (e.g., floor location, rack location, passive conveyor, active conveyor, etc.). <br>The load type informs about the load unit and can be used to switch field for example (e.g., EPAL, INDU, etc). <br>For preparing the load handling device (e.g., pre-lift operations based on the height parameter), the action could be announced in the horizon in advance. <br>But, pre-Lift operations, etc., are not reported as 'RUNNING' in the mobile robot state, because the associated node is not released yet.<br>If on an edge, the mobile robot can use its sensing device to detect the position for picking the node. | no | loadHandlingDevice (string, optional)<br>stationType (string, optional)<br>stationName (string, optional)<br>loadType (string, optional) <br>loadId (string, optional)<br>height (float64, optional)<br>defines bottom of the load related to the floor<br>depth (float64, optional) for forklifts<br>side (string, optional) e.g., conveyor | loads | no | yes | yes | no
```

drop | pick<br><br>(if automated) | Request the mobile robot to drop a load. <br>See action pick for more details. | no | loadHandlingDevice (string, optional)<br>stationType (string, optional)<br>stationName (string, optional)<br>loadType (string, optional)<br>loadId (string, optional)<br>height (float64, optional)<br>depth (float64, optional) <br>… | loads | no | yes | yes | no
detectObject | - | Mobile robot detects object (e.g., load, charging spot, free parking position). | yes | objectType (string, optional) | - | no | yes | yes | yes
finePositioning | - | On a node, mobile robot will position exactly on a target.<br>The mobile robot is allowed to deviate from its node position.<br>On an edge, the mobile robot will e.g., align on stationary equipment while traversing an edge. | yes | stationType (string, optional)<br>stationName (string, optional) | - | no | yes | yes | yes

> waitForTrigger | - | Mobile robot shall wait for a trigger of the type specified in the triggerType parameter, which is an array of strings. Two predefined values shall be used when semantically appropriate: 'FLEET_CONTROL' if the trigger originates from the fleet control, and 'LOCAL' if the trigger comes from an input on the mobile robot (e.g., button press, manual loading). If none of the predefined values meet the specific requirements, custom values can be defined. <br>Fleet control is responsible for handling the timeout and shall cancel the order if necessary. | yes | triggerType [string] (array) | - | no | yes | no | yes

Reviewer 3: ```suggestion
waitForTrigger | - | Mobile robot shall wait for a trigger of the type specified in the `triggerType` parameter, which is an array of strings. Two predefined values shall be used when semantically appropriate: 'FLEET_CONTROL' if the trigger originates from the fleet control, and 'LOCAL' if the trigger comes from an input on the mobile robot (e.g., button press, manual loading). If none of the predefined values meet the specific requirements, custom values can be defined. <br>Fleet control is responsible for handling the timeout and shall cancel the order if necessary. | yes | triggerType [string] (array) | - | no | yes | no | yes
```

> drop | pick<br><br>(if automated) | Request the mobile robot to drop a load. <br>See action pick for more details. | no | loadHandlingDevice (string, optional)<br>stationType (string, optional)<br>stationName (string, optional)<br>loadType (string, optional)<br>loadId (string, optional)<br>height (float64, optional)<br>depth (float64, optional) <br>… | .load | no | yes | yes | no

Reviewer: .load -> loads
```suggestion
pick | drop<br><br>(if automated) | Request the mobile robot to pick a load. <br>Mobile robots with multiple load handling devices can process multiple pick operations in parallel. <br>In this case, the parameter loadHandlingDevice needs to be present (e.g., LHD1). <br>The parameter stationType informs how the pick operation is handled in detail (e.g., floor location, rack location, passive conveyor, active conveyor, etc.). <br>The load type informs about the load unit and can be used to switch field for example (e.g., EPAL, INDU, etc). <br>For preparing the load handling device (e.g., pre-lift operations based on the height parameter), the action could be announced in the horizon in advance. <br>But, pre-Lift operations, etc., are not reported as 'RUNNING' in the mobile robot state, because the associated node is not released yet.<br>If on an edge, the mobile robot can use its sensing device to detect the position for picking the node. | no | loadHandlingDevice (string, optional)<br>stationType (string, optional)<br>stationName (string, optional)<br>loadType (string, optional) <br>loadId (string, optional)<br>height (float64, optional)<br>defines bottom of the load related to the floor<br>depth (float64, optional) for forklifts<br>side (string, optional) e.g., conveyor | loads | no | yes | yes | no
drop | pick<br><br>(if automated) | Request the mobile robot to drop a load. <br>See action pick for more details. | no | loadHandlingDevice (string, optional)<br>stationType (string, optional)<br>stationName (string, optional)<br>loadType (string, optional)<br>loadId (string, optional)<br>height (float64, optional)<br>depth (float64, optional) <br>… | loads | no | yes | yes | no
```

detectObject | - | Mobile robot detects object (e.g., load, charging spot, free parking position). | yes | objectType (string, optional) | - | no | yes | yes | yes
finePositioning | - | On a node, mobile robot will position exactly on a target.<br>The mobile robot is allowed to deviate from its node position.<br>On an edge, the mobile robot will e.g., align on stationary equipment while traversing an edge. | yes | stationType (string, optional)<br>stationName (string, optional) | - | no | yes | yes | yes

> waitForTrigger | - | Mobile robot shall wait for a trigger of the type defined specified in the triggerType parameter, which is an array of strings. Two predefined values shall be used when semantically appropriate: 'FLEET_CONTROL' if the trigger originates from the fleet control, and 'LOCAL' if the trigger comes from an input on the mobile robot (e.g., button press, manual loading). If none of the predefined values meet the specific requirements, custom values can be defined. <br>Fleet control is responsible for handling the timeout and shall cancel the order if necessary. | yes | triggerType [string] (array) | - | no | yes | no | yes

Reviewer: "specified" or "defined" here? would go for specified
```suggestion
waitForTrigger | - | Mobile robot shall wait for a trigger of the type specified in the triggerType parameter, which is an array of strings. Two predefined values shall be used when semantically appropriate: 'FLEET_CONTROL' if the trigger originates from the fleet control, and 'LOCAL' if the trigger comes from an input on the mobile robot (e.g., button press, manual loading). If none of the predefined values meet the specific requirements, custom values can be defined. <br>Fleet control is responsible for handling the timeout and shall cancel the order if necessary. | yes | triggerType [string] (array) | - | no | yes | no | yes
```

downloadZoneSet | Initialize the connection to the zone set server. | Mobile robot is downloading the zone set. | - | The download has finished. The mobile robot updates its state by setting a corresponding zoneSet object in its state with zoneSetStatus 'DISABLED'. | The download failed, updated in mobile robot state (e.g., connection lost, server unreachable, zone set not existing, zone set with same zoneSetId already on mobile robot). | Download failed or was interrupted. The mobile robot is waiting for intervention from fleet control.
enableZoneSet | - | Mobile robot enables the zone set with the requested zoneSetId and disables any other zone set for the same mapId. | - | The zone set has been enabled. The mobile robot updates the corresponding zoneSetStatus of the requested zoneSet to 'ENABLED' and the other zone sets for the same mapId to 'DISABLED'. | The requested zone set does not exist.| -

> deleteZoneSet | - | Mobile robot deletes the zone set with requested zoneSetId from its internal memory. | - | The zone set has been deleted. The mobile robot removes zoneSet object from its state. | The zone set could not be deleted, e.g., because zone set is currently in use or the requested zone set has already been deleted before. | -

Reviewer 3: ```suggestion
deleteZoneSet | - | Mobile robot deletes the zone set with requested `zoneSetId` from its internal memory. | - | The zone set has been deleted. The mobile robot removes `zoneSet` object from its state. | The zone set could not be deleted, e.g., because zone set is currently in use or the requested zone set has already been deleted before. | -
```

> clearInstantActions | - | | - | The instant actions array has been cleaned from all FINISHED or FAILED instantActions. | - | - 

Reviewer 3: ```suggestion
clearInstantActions | - | | - | The instant actions array has been cleaned from all 'FINISHED' or 'FAILED' instantActions. | - | - 
```

> clearZoneActions | - | | - | The zone actions array has been cleaned from all FINISHED or FAILED zoneActions. | - | - 

Reviewer 3: ```suggestion
clearZoneActions | - | | - | The zone actions array has been cleared off all 'FINISHED' or 'FAILED' zone actions. | - | - 
```

> | BLOCKED | none | | Mobile robots shall not enter this zone. If a mobile robot has entered the zone or finds itself within one, it shall stop and throw an 'BLOCKED_ZONE_VIOLATION' error with level set to 'CRITICAL'.| 

Reviewer: typo
```suggestion
| BLOCKED | none | | Mobile robots shall not enter this zone. If a mobile robot has entered the zone or finds itself within one, it shall stop and throw a 'BLOCKED_ZONE_VIOLATION' error with level set to 'CRITICAL'.| 
```

| LINE_GUIDED | none | | No free navigation is allowed in this zone, mobile robots shall follow the predefined trajectories on edges. Mobile robots may only enter this zone if the route is explicitly specified by the fleet control in the form of a node-edge graph. Any movement of the mobile robot that requires it to enter this zone shall follow a predefined trajectory. When entering the zone, the mobile robot shall be on the trajectory of the edge that crosses the zone. The edges that enter and are inside the line-guided zone require a trajectory sent from the fleet control or a predefined trajectory on the mobile robot. A corridor can be sent to allow the mobile robot to deviate from the trajectory. | 
| RELEASE | | - | Mobile robots are only allowed entering this zone once they have been granted access through fleet control. | 

> | | releaseLossBehavior | string | Enum {'STOP', 'CONTINUE', 'EVACUATE'}<br>Defines the behavior of the mobile robot when the release for this zone is revoked or expires while the mobile robot is inside the zone. If not defined, the mobile robot shall behave as if 'STOP' was defined.<br>'STOP': Mobile robot stops driving and reports a 'RELEASE_LOST' error with level 'CRITICAL'.<br>'EVACUATE': Mobile robot executes its evacuation behavior to leave the zone. It keeps the request in its state with the requestStatus set to 'REVOKED' or 'EXPIRED' accordingly and reports a 'RELEASE_LOSS_HANDLING' error with level 'WARNING' until the zone has been left.<br>'CONTINUE': Mobile robot continues its path. It keeps the request in its state with the requestStatus set to 'REVOKED' or 'EXPIRED' accordingly and reports a 'RELEASE_LOSS_HANDLING' error with level 'WARNING' until either the zone has been left or the order has ended inside the zone. If the order ends inside the zone, the mobile robot waits for a new order.|

Reviewer 2: Since an expected default value is defined, this parameter is supposed to be optional.
```suggestion
| | *releaseLossBehavior* | string | Enum {'STOP', 'CONTINUE', 'EVACUATE'}<br>Defines the behavior of the mobile robot when the release for this zone is revoked or expires while the mobile robot is inside the zone. If not defined, the mobile robot shall behave as if 'STOP' was defined.<br>'STOP': Mobile robot stops driving and reports a 'RELEASE_LOST' error with level 'CRITICAL'.<br>'EVACUATE': Mobile robot executes its evacuation behavior to leave the zone. It keeps the request in its state with the requestStatus set to 'REVOKED' or 'EXPIRED' accordingly and reports a 'RELEASE_LOSS_HANDLING' error with level 'WARNING' until the zone has been left.<br>'CONTINUE': Mobile robot continues its path. It keeps the request in its state with the requestStatus set to 'REVOKED' or 'EXPIRED' accordingly and reports a 'RELEASE_LOSS_HANDLING' error with level 'WARNING' until either the zone has been left or the order has ended inside the zone. If the order ends inside the zone, the mobile robot waits for a new order.|
```

> | | *releaseLossBehavior* | string | Enum {'STOP', 'CONTINUE', 'EVACUATE'}<br>Defines the behavior of the mobile robot when the release for this zone is revoked or expires while the mobile robot is inside the zone. If not defined, the mobile robot shall behave as if 'STOP' was defined.<br>'STOP': Mobile robot stops driving and reports a 'RELEASE_LOST' error with level 'CRITICAL'.<br>'EVACUATE': Mobile robot executes its evacuation behavior to leave the zone. It keeps the request in its state with the requestStatus set to 'REVOKED' or 'EXPIRED' accordingly and reports a 'RELEASE_LOSS_HANDLING' error with level 'WARNING' until the zone has been left.<br>'CONTINUE': Mobile robot continues its path. It keeps the request in its state with the requestStatus set to 'REVOKED' or 'EXPIRED' accordingly and reports a 'RELEASE_LOSS_HANDLING' error with level 'WARNING' until either the zone has been left or the order has ended inside the zone. If the order ends inside the zone, the mobile robot waits for a new order.|

Reviewer: backticks
```suggestion
| | *releaseLossBehavior* | string | Enum {'STOP', 'CONTINUE', 'EVACUATE'}<br>Defines the behavior of the mobile robot when the release for this zone is revoked or expires while the mobile robot is inside the zone. If not defined, the mobile robot shall behave as if 'STOP' was defined.<br>'STOP': Mobile robot stops driving and reports a 'RELEASE_LOST' error with level 'CRITICAL'.<br>'EVACUATE': Mobile robot executes its evacuation behavior to leave the zone. It keeps the request in its state with the `requestStatus` set to 'REVOKED' or 'EXPIRED' accordingly and reports a 'RELEASE_LOSS_HANDLING' error with level 'WARNING' until the zone has been left.<br>'CONTINUE': Mobile robot continues its path. It keeps the request in its state with the `requestStatus` set to 'REVOKED' or 'EXPIRED' accordingly and reports a 'RELEASE_LOSS_HANDLING' error with level 'WARNING' until either the zone has been left or the order has ended inside the zone. If the order ends inside the zone, the mobile robot waits for a new order.|
```

> | COORDINATED_REPLANNING | none | | No autonomous replanning is allowed within this zone. Mobile robots are only allowed adjusting their path if granted permission by fleet control. | 

Reviewer: At 6.4.3, behavior according to releaseLoss is described.
However, this parameter is not defined at all for COORDINATED_REPLANNING zone.

We should either add it or remove the behavior description.

Reviewer 2: We removed that for COORDINATED_REPLANNING, just stating the stop and the CRITICAL error being necessary, right?

> If the mobile robot receives a new zone set via the `zoneSet` topic or `downloadZoneSet` instant action with the same `zoneSetId` as an existing one, it shall not take over the zone set in its internal memory and report an error of type 'DUPLICATE_ZONE_SET' and level 'WARNING' for a reasonable amount of time for the fleet control to notice that the zone update failed.

Reviewer: missing shall  (ambiguous "shall not ..." otherwise)
```suggestion
If the mobile robot receives a new zone set via the `zoneSet` topic or `downloadZoneSet` instant action with the same `zoneSetId` as an existing one, it shall not take over the zone set in its internal memory and shall report an error of type 'DUPLICATE_ZONE_SET' and level 'WARNING' for a reasonable amount of time for the fleet control to notice that the zone update failed.
```

The `grantType` 'REVOKED' indicates that the permission is no longer valid. The fleet control shall assume a 'REVOKED' request as still being 'GRANTED', until the `requestStatus` of the mobile robot is set to 'REVOKED'.
The `response` object can include a `leaseExpiry` which specifies until when a 'GRANTED' request is valid. To extend the `leaseExpiry` fleet control can resend a response message with an updated `leaseExpiry` time.

The mobile robot shall acknowledge the fleet controls response by setting the `requestStatus` accordingly and keep the request for as long as it considers the information relevant. See also Section [6.9 Request/response mechanism](#69-requestresponse-mechanism).

The interaction between the mobile robot and the fleet control for 'RELEASE' zones shall be according to Figure 16.

While the mobile robot remains in the 'RELEASE' zone, it keeps the `zoneRequest` object in its state and continues to report `requestStatus` as 'GRANTED' to inform fleet control that it is still inside the zone. After mobile robot has exited the zone, it shall remove the corresponding `zoneRequest` entry from its state message.

> When receiving a response with `grantType` 'REVOKED', the mobile robot shall remove the request from its state. When the `leaseExpiry` has passed, the requestStatus shall be set to 'EXPIRED' and the zone shall not be entered. If the mobile robot is already inside the 'RELEASE' zone when the `leaseExpiry` has passed or the request is 'REVOKED', it shall report a warning and react according to the `releaseLossBehavior` defined in the zone definition.

Reviewer: We defined a critical error for this situation
```suggestion
When receiving a response with `grantType` 'REVOKED', the mobile robot shall remove the request from its state. When the `leaseExpiry` has passed, the requestStatus shall be set to 'EXPIRED' and the zone shall not be entered. If the mobile robot is already inside the 'RELEASE' zone when the `leaseExpiry` has passed or the request is 'REVOKED', it shall communicate this with an error of type 'RELEASE_LOST' and error level 'CRITICAL' and react according to the `releaseLossBehavior` defined in the zone definition.
```

Reviewer 2: Not in every case of the loss of a release an error is to be thrown (only with 'STOP' behavior). The definition of release loss behavior states the following:
| **Zone Type**| **Zone Parameters** | **Data type** | **Description** | 
| --- | --- | --- | --- |
| RELEASE | | - | Mobile robots are only allowed entering this zone once they have been granted access through fleet control. | 
| | releaseLossBehavior | string | Enum {'STOP', 'CONTINUE', 'EVACUATE'}<br>When the access to this zone is revoked or expired, the mobile robot can either 'STOP', 'CONTINUE', or 'EVACUATE' the zone. This action is only executed, when the mobile robot is already in the zone and the release expires or is revoked. If not defined, the mobile robot is expected to STOP and report an error.<br>'STOP': Mobile robot stops and sends a 'RELEASE_LOST' error with level 'CRITICAL'.<br>'EVACUATE': Execute the evacuation behavior of the mobile robot to leave the zone, keeping the `zoneRequest` object granting release in its state until the zone is left.<br>'CONTINUE': If the release is revoked or expires after the mobile robot has already entered the zone, the mobile robot continues its path, keeping the `zoneRequest` object granting the zone release in its state. If the order ends inside the zone, the mobile robot waits for a new order.|

![Figure 16 Zone request behavior for a RELEASE zone.](./assets/request_release_zone_access.png)
>Figure 16 - Zone request behavior for a RELEASE zone.

The interaction between the mobile robot and the fleet control for 'COORDINATED_REPLANNING' zones shall be according to Figure 17.

The mobile robot shall choose one of the trajectories of all 'GRANTED' requests to the zone and set the corresponding `requestStatus`to 'GRANTED' while removing all other requests from its state.

> When receiving a response with `grantType` 'REVOKED', the mobile robot shall remove the request from its state and not enter the 'COORDINATED_REPLANNING' zone. When the `leaseExpiry` has passed, the `requestStatus` shall be set to 'EXPIRED' and the zone shall not be entered. If the mobile robot is already inside the 'RELEASE' zone when the `leaseExpiry` has passed or the request is 'REVOKED', it shall stop driving and report a warning. To continue, the mobile robot shall state a new request.

Reviewer: wrong zone type for this section
```suggestion
When receiving a response with `grantType` 'REVOKED', the mobile robot shall remove the request from its state and not enter the 'COORDINATED_REPLANNING' zone. When the `leaseExpiry` has passed, the `requestStatus` shall be set to 'EXPIRED' and the zone shall not be entered. If the mobile robot is already inside the 'COORDINATED_REPLANNING' zone when the `leaseExpiry` has passed or the request is 'REVOKED', it shall stop driving and report a warning. To continue, the mobile robot shall state a new request.
```

> When receiving a response with `grantType` 'REVOKED', the mobile robot shall remove the request from its state and not enter the 'COORDINATED_REPLANNING' zone. When the `leaseExpiry` has passed, the `requestStatus` shall be set to 'EXPIRED' and the zone shall not be entered. If the mobile robot is already inside the 'RELEASE' zone when the `leaseExpiry` has passed or the request is 'REVOKED', it shall stop driving and report a warning. To continue, the mobile robot shall state a new request.

Reviewer: `... it shall stop driving and report a warning`
Should we also use the CRITICAL error 'RELEASE_LOST' here?

Reviewer 2: Here, the CRITICAL ERROR would be the correct behavior, since immediate attention of fleet control is required.

> When receiving a response with `grantType` 'REVOKED', the mobile robot shall remove the request from its state and not enter the 'COORDINATED_REPLANNING' zone. When the `leaseExpiry` has passed, the `requestStatus` shall be set to 'EXPIRED' and the zone shall not be entered. If the mobile robot is already inside the 'COORDINATED_REPLANNING' zone when the `leaseExpiry` has passed or the request is 'REVOKED', it shall stop driving and report a warning. To continue, the mobile robot shall state a new request.

Reviewer 2: According to proposal by @cpaulsynaos 
```suggestion
When receiving a response with `grantType` 'REVOKED', the mobile robot shall remove the request from its state and not enter the 'COORDINATED_REPLANNING' zone. When the `leaseExpiry` has passed, the `requestStatus` shall be set to 'EXPIRED' and the zone shall not be entered. If the mobile robot is already inside the 'COORDINATED_REPLANNING' zone when the `leaseExpiry` has passed or the request is 'REVOKED', it shall stop driving and it shall communicate this with an error of type 'RELEASE_LOST' and error level 'CRITICAL'. To continue, the mobile robot shall state a new request.
```

Fleet control responds to zone requests via the `responses` topic.

> The response message contains an array of `response` objects. Each `response` shall only respond to a single request referenced by the `requestId`.

Reviewer: plural
```suggestion
The responses message contains an array of `response` objects. Each `response` shall only respond to a single request referenced by the `requestId`.
```

Each response has a `grantType` that is either 'GRANTED', 'QUEUED', 'REVOKED', or 'REJECTED'.
If the `grantType` is 'GRANTED', the mobile robot is allowed to enter the zone or use the requested trajectory.
Fleet control can set the `grantType` to 'QUEUED' to acknowledge the mobile robot's request without giving permission, informing the mobile robot that its request is being processed.
If the `grantType` is 'REJECTED', the mobile robot shall not enter the zone or use the requested trajectory.
The `grantType` 'REVOKED' indicates that the permission is no longer valid. The fleet control shall assume a 'REVOKED' request as still being 'GRANTED', until the `requestStatus` of the mobile robot is set to 'REVOKED'.

> The `response` object can include a `leaseExpiry` which specifies until when a 'GRANTED' request is valid. To extend the `leaseExpiry` fleet control can resend a response message with an updated `leaseExpiry` time.

Reviewer: plural
```suggestion
The `response` object can include a `leaseExpiry` which specifies until when a 'GRANTED' request is valid. To extend the `leaseExpiry` fleet control can resend a responses message with an updated `leaseExpiry` time.
```

> The mobile robot shall acknowledge the fleet controls response by setting the `requestStatus` accordingly and keep the request for as long as it considers the information relevant. See also Section [6.9 Request/response mechanism](#69-requestresponse-mechanism).

Reviewer: grammar
```suggestion
The mobile robot shall acknowledge the fleet control's response by setting the `requestStatus` accordingly and keep the request for as long as it considers the information relevant. See also Section [6.9 Request/response mechanism](#69-requestresponse-mechanism).
```

The interaction between the mobile robot and the fleet control for 'RELEASE' zones shall be according to Figure 16.

> While the mobile robot remains in the 'RELEASE' zone, it keeps the `zoneRequest` object in its state and continues to report `requestStatus` as 'GRANTED' to inform fleet control that it is still inside the zone. After mobile robot has exited the zone, it shall remove the corresponding `zoneRequest` entry from its state message.

Reviewer: missing the
```suggestion
While the mobile robot remains in the 'RELEASE' zone, it keeps the `zoneRequest` object in its state and continues to report `requestStatus` as 'GRANTED' to inform fleet control that it is still inside the zone. After the mobile robot has exited the zone, it shall remove the corresponding `zoneRequest` entry from its state message.
```

If the mobile robot has not yet entered the 'RELEASE' zone and either a response with `grantType` 'REVOKED' is received or the `leaseExpiry` has passed, it shall remove the request from its state and shall not enter the zone.
If the mobile robot is already inside the 'RELEASE' zone when a response with `grantType` 'REVOKED' is received or the `leaseExpiry` has passed, it shall react according to the `releaseLossBehavior` defined in the zone definition. It shall keep the request in its state with the `requestStatus` set to 'REVOKED' or 'EXPIRED' accordingly, until the zone has been left or a new release is granted. To re-enter the zone, the mobile robot shall issue a new request.

![Figure 16 Zone request behavior for a RELEASE zone.](./assets/request_release_zone_access.png)
>Figure 16 - Zone request behavior for a RELEASE zone.

The interaction between the mobile robot and the fleet control for 'COORDINATED_REPLANNING' zones shall be according to Figure 17.

> The mobile robot shall choose one of the trajectories of all 'GRANTED' requests to the zone and set the corresponding `requestStatus`to 'GRANTED' while removing all other requests from its state.

Reviewer: missing blank
```suggestion
The mobile robot shall choose one of the trajectories of all 'GRANTED' requests to the zone and set the corresponding `requestStatus` to 'GRANTED' while removing all other requests from its state.
```

'START_NODE_OUT_OF_RANGE' | 'WARNING' | Receival of an order with unreachable first node. | `orderId` | Until new order is accepted.
'MOBILE_ROBOT_NOT_AVAILABLE' | 'WARNING' | Receival of an order while not in 'AUTOMATIC', 'SEMIAUTOMATIC' or 'INTERVENED' operating mode. | `orderId` | Until operating mode allows for new orders
'UNKNOWN_MAP_ID' | 'WARNING' | Receival of an order containing nodes referencing an unknown `mapId`. | `orderId` | Until new order is accepted.
'INSTANT_ACTION_STATES_FULL' | 'URGENT' | Array of `instantActionStates` is becoming too long to manage | - | Until instantActionStates are cleared by fleet control.

> ZONE_ACTION_STATES_FULL' | 'URGENT' | Array of `zoneActionStates` is becoming too long to manage | - | Until instantActionStates are cleared by fleet control.

Reviewer 2: ```suggestion
'ZONE_ACTION_STATES_FULL' | 'URGENT' | Array of `zoneActionStates` is becoming too long to manage | - | Until instantActionStates are cleared by fleet control.
```

'START_NODE_OUT_OF_RANGE' | 'WARNING' | Receival of an order with unreachable first node. | `orderId` | Until new order is accepted.
'MOBILE_ROBOT_NOT_AVAILABLE' | 'WARNING' | Receival of an order while not in 'AUTOMATIC', 'SEMIAUTOMATIC' or 'INTERVENED' operating mode. | `orderId` | Until operating mode allows for new orders
'UNKNOWN_MAP_ID' | 'WARNING' | Receival of an order containing nodes referencing an unknown `mapId`. | `orderId` | Until new order is accepted.
'INSTANT_ACTION_STATES_FULL' | 'URGENT' | Array of `instantActionStates` is becoming too long to manage | - | Until instantActionStates are cleared by fleet control.

> 'ZONE_ACTION_STATES_FULL' | 'URGENT' | Array of `zoneActionStates` is becoming too long to manage | - | Until instantActionStates are cleared by fleet control.

Reviewer: Typo 
```suggestion
'ZONE_ACTION_STATES_FULL' | 'URGENT' | Array of `zoneActionStates` is becoming too long to manage | - | Until zoneActionStates are cleared by fleet control.
```

Reviewer 2: done in next comment

> 'ZONE_ACTION_STATES_FULL' | 'URGENT' | Array of `zoneActionStates` is becoming too long to manage | - | Until instantActionStates are cleared by fleet control.

Reviewer: missing period
```suggestion
'INSTANT_ACTION_STATES_FULL' | 'URGENT' | Array of `instantActionStates` is becoming too long to manage. | - | Until instantActionStates are cleared by fleet control.
'ZONE_ACTION_STATES_FULL' | 'URGENT' | Array of `zoneActionStates` is becoming too long to manage. | - | Until zoneActionStates are cleared by fleet control.
```

'DUPLICATE_MAP' | 'WARNING' | Receival of a map with `mapId` and `mapVersion` already existing. | `mapId` and `mapVersion` of duplicate | Until a new map related instantAction was accepted.
'BLOCKED_ZONE_VIOLATION' | 'CRITICAL' | Entering a 'BLOCKED' zone. | `zoneId` | Until the mobile robot is no longer violating the blocked zone.
'DUPLICATE_ZONE_SET' | 'WARNING' | Receival of a zone set with `zoneSetId` already existing. | `zoneSetId` or `actionId` of `instantAction` | Reasonable amount of time for the fleet control to notice that the zone update failed.
'RELEASE_LOSS_HANDLING' | 'WARNING' | Executing the 'EVACUATE' or 'CONTINUE' release loss behavior after losing the release for a 'RELEASE' zone while the mobile robot is inside the zone. | `zoneId` | Until the mobile robot has left the zone, is granted the release again, or the order has ended inside the zone.

> 'RELEASE_LOST' | 'CRITICAL' | Stopping within a zone due to losing the release for a 'RELEASE' or 'COORDINATED_REPLANNING' zone while the mobile robot is inside the zone. | `zoneId` | Until the mobile robot has left the zone or is granted the release again.

Reviewer 3: ```suggestion
'RELEASE_LOST' | 'CRITICAL' | Stopping within a zone due to losing the release for a 'RELEASE' zone with undefined or 'STOP' release loss behavior or 'COORDINATED_REPLANNING' zone while the mobile robot is inside the zone. | `zoneId` | Until the mobile robot has left the zone or is granted the release again.
```

Reviewer 2: STOP is the default expected behavior when not defined. Do we want to make this explicit like proposed here?

Reviewer 2: I would only use the hint on STOP and extend it for corridors as well in another PR

'WAITING' | Action was received by the mobile robot but the corresponding node was not yet traversed or the corresponding edge was not yet entered.
'INITIALIZING' | Action was triggered, preparatory measures are initiated.
'RUNNING' | The action is running.

> 'PAUSED' | The action is paused because of a startPause instantAction or external trigger (pause button on the mobile robot)

Reviewer 3: ```suggestion
'PAUSED' | The action is paused because of a `startPause` instant action or external trigger (pause button on the mobile robot)
```

> 'RETRIABLE' | Actions that failed, but can be retried, specified by the retriable parameter in the action of an order. Transition from this state is triggered by a retry or skipRetry instantAction or an external trigger.

Reviewer 3: ```suggestion
'RETRIABLE' | Actions that failed, but can be retried, specified by the `retriable` parameter in the action of an order. Transition from this state is triggered by a `retry` or `skipRetry` instant action or an external trigger.
```

'FINISHED' | The action is finished. <br>A result is reported via the `actionResult`.

> 'FAILED' | Action could not be finished for whatever reason.

Reviewer 3: ```suggestion
'FAILED' | Action could not be finished.
```

If the corridors within a mobile robot's currently active order have the `releaseRequired` flag set to true, it shall issue a request prior to deviating from the predefined trajectory of an edge. For this purpose, the robot shall add an `edgeRequest` object to its state message. The `requestId` shall be unique across all requests (e.g., `zoneRequest`, `edgeRequest`) issued by the mobile robot.

> The `requestStatus` is set to REQUESTED and the combination of `edgeId` and `sequenceId` references the edge's trajectory the robot asks to deviate from. The mobile robot has the option to request the approval for several edges simultaneously as long as they are part of its current base. The usage of each corridor shall be requested in a dedicated `edgeRequest` and each request shall be approved individually by fleet control via the `response` topic (see Section [6.9 Request/response mechanism](#69-requestresponse-mechanism)).

Reviewer: plural
```suggestion
The `requestStatus` is set to REQUESTED and the combination of `edgeId` and `sequenceId` references the edge's trajectory the robot asks to deviate from. The mobile robot has the option to request the approval for several edges simultaneously as long as they are part of its current base. The usage of each corridor shall be requested in a dedicated `edgeRequest` and each request shall be approved individually by fleet control via the `responses` topic (see Section [6.9 Request/response mechanism](#69-requestresponse-mechanism)).
```

If a variable is marked as optional, it is optional for the sender as the variable might not be applicable in certain cases (e.g., when the fleet control sends an order to a mobile robot, some mobile robots plan their trajectory themselves and the field `trajectory` within the `edge` object of the order can be omitted).

If the mobile robot receives a message that contains a field which is marked as optional in this protocol, the mobile robot is expected to act accordingly and shall not ignore the field.

> If the mobile robot cannot process the order due to an unsupported parameter, it shall communicate this with an error of type 'UNSUPPORTED_PARAMETER' and error level 'CRITICAL' and reject the order.

Reviewer 3: ```suggestion
If the mobile robot cannot process the order due to an unsupported parameter, it shall communicate this with an error of type 'UNSUPPORTED_PARAMETER' and level 'CRITICAL' and reject the order.
```

x | m | float64 | X-position on the map in reference to the global project-specific coordinate system. <br>Precision is up to the specific implementation.
y | m | float64 | Y-position on the map in reference to the global project-specific coordinate system. <br>Precision is up to the specific implementation.
*theta* | rad | float64 | Range: [-Pi ... Pi] <br><br>Absolute orientation a mobile robot shall match on a node for it to be considered traversed.<br>If defined, the mobile robot shall match the orientation on this node.<br>If previous edge disallows rotation, the mobile robot shall rotate on the node.<br>If following edge has a differing orientation defined but disallows rotation, the mobile robot shall rotate on the node to the edges desired rotation before entering the edge.

> ***allowedDeviationXY*** | m | JSON object | Indicates how precisely a mobile robot shall match the position of a node for it to be considered traversed.<br>(see also Section [6.1.3 Order cancellation](#613-order-cancellation) and [6.6.2 Traversal of nodes and edges](#662-traversal-of-nodes-and-edges)).

Reviewer: Unrelated section
```suggestion
***allowedDeviationXY*** | m | JSON object | Indicates how precisely a mobile robot shall match the position of a node for it to be considered traversed.<br>(see also Section [6.6.2 Traversal of nodes and edges](#662-traversal-of-nodes-and-edges)).
```

*maximumSpeed* | m/s | float64 | Permitted maximum speed on the edge. <br>Speed is defined by the fastest measurement of the mobile robot.
*maximumMobileRobotHeight* | m | float64 | Permitted maximum height of the mobile robot, including the load, on the edge.
*minimumLoadHandlingDeviceHeight* | m | float64 | Permitted minimal height of the load handling device on the edge.

> *orientation* | rad | float64 | Orientation of the mobile robot on the trajectory of the edge. The value `orientationType` defines whether it shall be interpreted relative to the global project-specific map coordinate system or tangential to the trajectory of the edge. In case of tangential to the trajectory, 0.0 denotes driving forwards and PI denotes driving backwards. <br>Example: orientation Pi/2 rad may lead to a rotation of 90 degrees.<br><br>If the mobile robot starts in a different orientation, and if `reachOrientationBeforeEntering` is set to "false", rotate the mobile robot on the edge to the desired orientation.<br>If `reachOrientationBeforeEntering` is "true", rotate before entering the edge.<br>If this is not possible, the order shall be rejected.<br><br>If no trajectory is defined, apply the orientation and any rotation to the direct path between the two connecting nodes of the edge.<br>If no orientation is defined, the mobile robot may assume any orientation on the edge.

Reviewer 3: Should we add a predefined errorType the robot shall throw when rejecting an orderUpdate for this reasons (cannot rotate on spot)?

Reviewer 2: Done in PR #658: Added predefined error type 'ORIENTATION_UNREACHABLE' ('WARNING', reference `edgeId`) for rejecting an order or order update with an edge `orientation` the mobile robot cannot reach (new section 6.1.4.11).

| response <br> { | JSON object | Object which contains the fleet control's answer to a specific request. |
| requestId | string | Unique per mobile robot identifier within all active requests. |

> | grantType | enum | Enum {'GRANTED','QUEUED','REVOKED','REJECTED'}<br>'GRANTED': The fleet control has granted the request. 'REVOKED': The fleet control revokes previously granted request. 'REJECTED': The Fleet control rejects a request. 'QUEUED': Acknowledge the mobile robot's request to the fleet control, but no permission is given yet. Request was added to some sort of a queue. |

Reviewer: cap + enum order + breaks
```suggestion
| grantType | enum | Enum {'GRANTED','QUEUED','REVOKED','REJECTED'}<br>'GRANTED': The fleet control has granted the request.<br> 'QUEUED': Acknowledge the mobile robot's request to the fleet control, but no permission is given yet. Request was added to some sort of a queue.<br> 'REVOKED': The fleet control revokes previously granted request.<br> 'REJECTED': The Fleet control rejects a request. |
```

> | *leaseExpiry* <br><br> } | string | Timestamp (ISO 8601, UTC); YYYY-MM-DDTHH:mm:ss.fffZ (e.g.“2017-04-15T11:40:03.123Z”). A timestamp for the release to expire shall only be sent with responses granting a request.

Reviewer: missing ,
```suggestion
| *leaseExpiry* <br><br> } | string | Timestamp (ISO 8601, UTC); YYYY-MM-DDTHH:mm:ss.fffZ (e.g., “2017-04-15T11:40:03.123Z”). A timestamp for the release to expire shall only be sent with responses granting a request.
```

Object structure | Unit | Data type | Description
---|---|---|---
**zoneSet** { | | JSON object|

> zoneSetId | | string | Unique identifier of the zone set that is currently enabled for the map.<br> This field shall be left empty only if the mobile robot has no zones defined for the corresponding map.

Reviewer 2: ```suggestion
zoneSetId | | string | Unique identifier of the zone set.
```

**mobileRobotPosition** { | | JSON object | Defines the position on a map in project-specific coordinates. Each floor has its own map.
x | m | float64 | X-position on the map in reference to the project-specific coordinate system. <br>Precision is up to the specific implementation.
y | m | float64 | Y-position on the map in reference to the project-specific coordinate system. <br>Precision is up to the specific implementation.
theta | rad | float64 | Range: [-Pi ... Pi]<br><br>Orientation of the mobile robot.
mapId | | string | Unique identification of the map in which the position is referenced.<br><br>Each map has the same origin of coordinates.<br>When a mobile robot uses an elevator from a departure floor to a destination floor, it leaves the map of the departure floor and spawns on the corresponding elevator node on the map of the destination floor.

> localized | | boolean | "true": Mobile robot is localized. `x`, `y`, and `theta` can be trusted.<br>"false": Mobile robot is not localized. `x`, `y`, and `theta` cannot be trusted.<br>Changing the state to "false" shall only happen if the mobile robot cannot determine its position anymore. The mobile robot shall report this state via an error (`errorType` = 'LOCALIZATION_ERROR', `errorLevel` = 'FATAL'). While this is set to "false", the mobile robot shall not resume automatic driving or continue its order.

Reviewer 3: ```suggestion
localized | | boolean | "true": Mobile robot is localized. `x`, `y`, and `theta` can be trusted.<br>"false": Mobile robot is not localized. `x`, `y`, and `theta` cannot be trusted.<br>Changing this parameter to "false" shall only happen if the mobile robot cannot determine its position anymore. The mobile robot shall report this via an error (`errorType` = 'LOCALIZATION_ERROR', `errorLevel` = 'FATAL'). While this is set to "false", the mobile robot shall not resume automatic driving or continue its order.
```

Reviewer 2: How about this parameter?

Object structure | Unit | Data type | Description
---|---|---|---
**error** { | | JSON object |

> errorType | | string | Error type, extensible enumeration including the following predefined values <br>Enum {'UNSUPPORTED_PARAMETER', 'NO_ORDER_TO_CANCEL', 'VALIDATION_FAILURE', 'INVALID_ORDER_ACTION', 'OUTDATED_ORDER_UPDATE', 'SAME_ORDER_UPDATE_ID', 'ORDER_UPDATE_FOLLOWING_CANCEL', 'OUTSIDE_OF_CORRIDOR', 'INSUFFICIENT_MEMORY', 'DUPLICATE_MAP', 'BLOCKED_ZONE_VIOLATION', 'DUPLICATE_ZONE_SET', 'RELEASE_LOSS_HANDLING','RELEASE_LOST', 'ZONE_ACTION_CONFLICT', 'NODE_UNREACHABLE', 'LOCALIZATION_ERROR', 'UNKNOWN_MAP_ID', 'OTHER_ORDER_ACTIVE', 'START_NODE_OUT_OF_RANGE', 'MOBILE_ROBOT_NOT_AVAILABLE', 'UNKNOWN_MAP_ID', 'INSTANT_ACTION_STATES_FULL', 'ZONE_ACTION_STATES_FULL', ...}.

Reviewer: double entry
```suggestion
errorType | | string | Error type, extensible enumeration including the following predefined values <br>Enum {'UNSUPPORTED_PARAMETER', 'NO_ORDER_TO_CANCEL', 'VALIDATION_FAILURE', 'INVALID_ORDER_ACTION', 'OUTDATED_ORDER_UPDATE', 'SAME_ORDER_UPDATE_ID', 'ORDER_UPDATE_FOLLOWING_CANCEL', 'OUTSIDE_OF_CORRIDOR', 'INSUFFICIENT_MEMORY', 'DUPLICATE_MAP', 'BLOCKED_ZONE_VIOLATION', 'DUPLICATE_ZONE_SET', 'RELEASE_LOSS_HANDLING','RELEASE_LOST', 'ZONE_ACTION_CONFLICT', 'NODE_UNREACHABLE', 'LOCALIZATION_ERROR', 'OTHER_ORDER_ACTIVE', 'START_NODE_OUT_OF_RANGE', 'MOBILE_ROBOT_NOT_AVAILABLE', 'UNKNOWN_MAP_ID', 'INSTANT_ACTION_STATES_FULL', 'ZONE_ACTION_STATES_FULL', ...}.
```
